python/tk_desktop/desktop_window.py

```diff
@@ -278,6 +291,48 @@ def __init__(self, parent=None):
         # be done only when the dialog is fully initialized.
         self._load_settings()
 
+    def _debug_toggled(self, state):
+        """
+        Toggles global debug logging and stores the given state as an
+        engine level user setting.
+
+        :param bool state: The debug state to set.
+        """
+        self.user_preferred_debug_logging = state
+        sgtk.LogManager().global_debug = state
+        sgtk.platform.current_engine().set_global_debug(state)
```

Comment:
This will fail automatically when you are not inside a project, which will yield a warning in the console that is not warranted.

Reply:
Resolved by adding an is_connected check lower down in the site engine implementation.